gamma round #2 updates

ixcat opened the pull request:
includes changes from 2021-04-23 discussion:

- [X] auth-free fetching
- [X] progress display 
  (only shows '1/N' since list objects is not an iterable with 'len'; 
   could convert to list then download but this might slow large-number-of-files archives significantly)
- [X] add errors for non-existing datasets for the 'revisions' and 'download' commands

kabilar commented:
Hi @ixcat , great job with this tool.  A few minor comments noted below:

- Authentication-free downloading works seamlessly.
- Can the progress display be with regards to the dataset size?
- If I enter the incorrect revision to download, the output is `dataset <dataset name> not found`.  Alternatively, it would be helpful to have the output be `revision <revision name> not found`.
- For the commands `djarchive datasets` and `djarchive revisions` it would helpful to have the print out organized in columns with column headings.
- For the command `djarchive revisions`, it would be helpful to also print out the dataset size.

ixcat commented:
deferred per above feedback:

- Q: Can the progress display be with regards to the dataset size? 
  A: need to precompute datasets to have file metadata available w/o extra IPC overhead / delays; 
      will defer.
- Q: For the command djarchive revisions, it would be helpful to also print out the dataset size.
  A: as above (need precomputed metadata)

updates per above & other feedback:

- Q: If I enter the incorrect revision to download, the output is dataset <dataset name> not found. Alternatively, it would be helpful to have the output be revision <revision name> not found.
  A: included name + revision in update

- Q: For the commands djarchive datasets and djarchive revisions it would helpful to have the print out organized in columns with column headings.
  A: Added simple print statement headings prior to listing output. Not done 'smartly' (e.g. heading will print before 'file not found' when datasets don't exist)

- Progress bar 'utility': due to limited utility of metadata-free tqdm(), switched 'display_progress' to simply print each file being downloaded as it happens.
